frame/staking/src/lib.rs

```diff
@@ -732,6 +731,143 @@ impl<AccountId, Balance: Default + HasCompact> Default for Exposure<AccountId, B
+	pub others: Vec<IndividualExposure<AccountId, Balance>>,
+}
+
+impl<AccountId, Balance: Default + HasCompact> Default for ExposurePage<AccountId, Balance> {
+	fn default() -> Self {
+		Self { page_total: Default::default(), others: vec![] }
+	}
+}
+
+/// An overview of stake backing a single validator.
+///
+/// It, in combination with a list of `ExposurePage`s, can be used to reconstruct a full `Exposure`
+/// struct. This is useful for cases where we want to query a single page of `Exposure`s.
+#[derive(PartialEq, Eq, PartialOrd, Ord, Clone, Encode, Decode, RuntimeDebug, TypeInfo)]
+pub struct ExposureOverview<Balance: HasCompact> {
+	/// The total balance backing this validator.
+	#[codec(compact)]
+	pub total: Balance,
+	/// The validator's own stash that is exposed.
+	#[codec(compact)]
+	pub own: Balance,
+	/// Number of nominators backing this validator.
+	pub nominator_count: u32,
+	/// Number of pages of backers.
+	pub page_count: PageIndex,
+}
+
+impl<Balance: Default + HasCompact> Default for ExposureOverview<Balance> {
+	fn default() -> Self {
+		Self {
+			total: Default::default(),
+			own: Default::default(),
+			nominator_count: Default::default(),
+			page_count: Default::default(),
+		}
+	}
+}
+
+/// Extended view of Exposure comprising of `ExposureOverview` and a single page of `ExposurePage`.
+///
+/// This is useful where we need to take into account the validator's own stake and total exposure
+/// in consideration in addition to the individual nominators backing them.
+#[derive(Encode, Decode, RuntimeDebug, TypeInfo, PartialEq, Eq)]
+struct ExposureExt<AccountId, Balance: HasCompact> {
+	exposure_overview: ExposureOverview<Balance>,
+	exposure_page: ExposurePage<AccountId, Balance>,
+}
+
+impl<AccountId, Balance: HasCompact + Copy + AtLeast32BitUnsigned> ExposureExt<AccountId, Balance> {
+	/// Create a new instance of `ExposureExt` from legacy clipped exposures.
+	pub fn from_clipped(exposure: Exposure<AccountId, Balance>) -> Self {
+		Self {
+			exposure_overview: ExposureOverview {
+				total: exposure.total,
+				own: exposure.own,
+				nominator_count: exposure.others.len() as u32,
+				page_count: 1,
+			},
+			exposure_page: ExposurePage { page_total: exposure.total, others: exposure.others },
+		}
+	}
+
+	/// Returns total exposure of this validator across pages
+	pub fn total(&self) -> Balance {
+		self.exposure_overview.total
+	}
+
+	/// Returns total exposure of this validator for current page
+	pub fn current_total(&self) -> Balance {
+		self.exposure_page.page_total + self.exposure_overview.own
+	}
+
+	/// Returns validator's own stake that is exposed
+	pub fn own(&self) -> Balance {
+		self.exposure_overview.own
+	}
+
+	/// Returns the portions of nominators stashes that are exposed in this page.
+	pub fn others(&self) -> &Vec<IndividualExposure<AccountId, Balance>> {
+		&self.exposure_page.others
+	}
+
+	/// Returns the number of pages of nominators stashes that are exposed.
+	#[allow(dead_code)]
```

Comment:
How so?

Reply:
This was used in tests only. Should replace it with `#[cfg(test)]`.

Reply:
If it's only used in tests, I'd probably just create a helper function in `mock.rs` to keep it clean.

Reply:
Yes right. But I think it will be used in the runtime api for page counts which is not added yet.